api/data.go

```diff
@@ -0,0 +1,72 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/ONSdigital/dp-legacy-cache-api/models"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+func (api *API) GetDataSets(ctx context.Context) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		log.Info(ctx, "Calling get datasets handler")
+
+		results, err := api.dataStore.GetDataSets(ctx)
+		if err != nil {
+			log.Error(ctx, "Error retrieving datasets from db:", err)
+			return
+		}
+
+		// Setting the header and encoding the results to JSON
+		w.Header().Set("Content-Type", "application/json; charset=utf-8")
+		err = json.NewEncoder(w).Encode(results)
+		if err != nil {
+			log.Error(ctx, "Error encoding results to JSON: %v", err)
+			http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+			return
+		}
+	}
+}
+
+// HelloHandler returns function containing a simple hello world example of an api handler
```

Comment:
I would remove this comment, since it's incorrect, and even if it was correct, it wouldn't add much value.